Add notifier for pagerduty (#31207)
mistercrunch/Airflow: docs/apache-airflow-providers-pagerduty/notifications/pagerduty_notifier_howto_guide.rst

 .. Licensed to the Apache Software Foundation (ASF) under one
    or more contributor license agreements.  See the NOTICE file
    distributed with this work for additional information
    regarding copyright ownership.  The ASF licenses this file
    to you under the Apache License, Version 2.0 (the
    "License"); you may not use this file except in compliance
    with the License.  You may obtain a copy of the License at

 ..   http://www.apache.org/licenses/LICENSE-2.0

 .. Unless required by applicable law or agreed to in writing,
    software distributed under the License is distributed on an
    "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
    KIND, either express or implied.  See the License for the
    specific language governing permissions and limitations
    under the License.

How-to Guide for Pagerduty notifications
========================================

Introduction
------------
Pagerduty notifier (:class:`airflow.providers.pagerduty.notifications.pagerduty.PagerdutyNotifier`) allows users to send

> messages to a pagerduty using the various ``on_*_callbacks`` at both the DAG level and Task level

Reviewer: ```suggestion
The Pagerduty notifier (:class:`airflow.providers.pagerduty.notifications.pagerduty.PagerdutyNotifier`) allows users to send
messages to Pagerduty using the various ``on_*_callbacks`` at both the DAG level and Task level.
```

Example Code:
-------------

.. code-block:: python

    from datetime import datetime
    from airflow import DAG
    from airflow.operators.bash import BashOperator
    from airflow.providers.pagerduty.notifications.notifier import send_pagerduty_notification

>     with DAG(dag_id="pagerduty_notifier", schedule_interval=None, start_date=datetime(2023, 1, 1), catchup=False):

Reviewer: ```suggestion
    with DAG("pagerduty_notifier", start_date=datetime(2023, 1, 1)):
```

Let's keep it simple.

Also, should we have a DAG level example as well?

Author: @jedcunningham Added a similar example on DAG level, please check

        BashOperator(
            task_id="mytask",

>             bash_command="redis-cli ping",

Reviewer: ```suggestion
            bash_command="fail",
```

    with the License.  You may obtain a copy of the License at

 ..   http://www.apache.org/licenses/LICENSE-2.0

 .. Unless required by applicable law or agreed to in writing,
    software distributed under the License is distributed on an
    "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
    KIND, either express or implied.  See the License for the
    specific language governing permissions and limitations
    under the License.

How-to Guide for Pagerduty notifications
========================================

Introduction
------------
The Pagerduty notifier (:class:`airflow.providers.pagerduty.notifications.pagerduty.PagerdutyNotifier`) allows users to send
messages to Pagerduty using the various ``on_*_callbacks`` at both the DAG level and Task level.

Example Code:
-------------

.. code-block:: python

    from datetime import datetime
    from airflow import DAG
    from airflow.operators.bash import BashOperator
    from airflow.providers.pagerduty.notifications.notifier import send_pagerduty_notification

    with DAG(
        "pagerduty_notifier",
        start_date=datetime(2023, 1, 1),
        on_failure_callback=[
            send_pagerduty_notification(

>                 summary="The task {{ ti.task_id }} failed",

Reviewer: ```suggestion
                summary="The dag {{ dag.dag_id }} failed",
```

 .. Unless required by applicable law or agreed to in writing,
    software distributed under the License is distributed on an
    "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
    KIND, either express or implied.  See the License for the
    specific language governing permissions and limitations
    under the License.

How-to Guide for Pagerduty notifications
========================================

Introduction
------------
Pagerduty notifier (:class:`airflow.providers.pagerduty.notifications.pagerduty_notifier.PagerdutyNotifier`) allows users to send
messages to a pagerduty using the various ``on_*_callbacks`` at both the DAG level and Task level

Example Code:
-------------

.. code-block:: python

    from datetime import datetime
    from airflow import DAG
    from airflow.operators.bash import BashOperator
    from airflow.providers.pagerduty.notifications.pagerduty_notifier import send_pagerduty_notification

    with DAG(
        dag_id="pagerduty_notifier",
        schedule_interval=None,
        start_date=datetime(2023, 1, 1),
        catchup=False,
        on_success_callback=[
            send_pagerduty_notification(

>                 summary="DISK at 99%",

Reviewer 2: Lets come up with a more realistic example attributes here please

Author: @ashb Changed the example, is this example fine?
